Sources/MockService.swift

```diff
@@ -10,76 +10,166 @@ open class MockService: NSObject {
     return interaction
   }
 
+  ///
+  /// Describe the request your provider will receive
+  ///
+  /// Use this method in the `Arrange` step of your Pact test.
+  ///
+  ///     myMockService.given("a user exists")
+  ///                  .uponReceiving("a request for users")
+  ///
+  /// - Parameter description: Describing the request to the provider
+  /// - Returns: An `Interaction` object
+  ///
   @objc(uponReceiving:)
   open func uponReceiving(_ description: String) -> Interaction {
     let interaction = Interaction().uponReceiving(description)
     interactions.append(interaction)
     return interaction
   }
 
+  ///
+  /// Runs the provided test function with 30 second timeout
+  ///
+  /// Use this method in the `Act` step of your Pact test.
+  /// (eg. Testing your `serviceClientUnderTest!.getUsers(...)` method)
+  ///
+  ///     myMockService!.run { (testComplete) -> Void in
+  ///         serviceClientUnderTest!.getUsers( /* ... */ )
+  ///     }
+  ///
+  /// Make sure you call `testComplete()` after your `Assert` step in your test
+  ///
+  /// - Parameter testFunction: The function making the network request you are testing
+  ///
   @objc(run:)
   open func objcRun(_ testFunction: @escaping (_ testComplete: () -> Void) -> Void) {
     self.run(nil, line: nil, timeout: 30, testFunction: testFunction)
   }
 
+  ///
+  /// Runs the provided test function by specifying timeout in seconds
+  ///
+  /// Use this method in the `Act` step of your Pact test.
+  /// (eg. Testing your `serviceClientUnderTest!.getUsers(...)` method)
+  ///
+  ///     myMockService!.run(timeout: 10) { (testComplete) -> Void in
+  ///         serviceClientUnderTest!.getUsers( /* ... */ )
+  ///     }
+  ///
+  /// Make sure you call `testComplete()` after your `Assert` step in your test
+  ///
+  /// - Parameter testFunction: The function making the network request you are testing
+  /// - Parameter timeout: Time to wait for the `testComplete()` else it fails the test
+  ///
   @objc(run: withTimeout:)
   open func objcRun(_ testFunction: @escaping (_ testComplete: () -> Void) -> Void,
                     timeout: TimeInterval) {
     self.run(nil, line: nil, timeout: timeout, testFunction: testFunction)
   }
 
-  open func run(_ file: FileString? = #file, line: UInt? = #line, timeout: TimeInterval = 30,
-                testFunction: @escaping (_ testComplete: @escaping () -> Void) -> Void) {
+  ///
+  /// Runs the provided test function
+  ///
+  /// Use this method in the `Act` step of your Pact test.
+  /// (eg. Testing your `serviceClientUnderTest!.getUsers(...)` method)
+  ///
+  ///     myMockService!.run(line: 40, timeout: 10) { (testComplete) -> Void in
+  ///         serviceClientUnderTest!.getUsers( /* ... */ )
+  ///     }
+  ///
+  /// Make sure you call `testComplete()` after your `Assert` step in your test
+  ///
+  /// - Parameter file: The file in which your test resides
+  /// - Parameter line: The line number of your test
```

Comment:
Probably don't want to document the `file` and `line` parameters. They are filled in by the compiler with the file and line of the calling method, which we then use in failWithLocation to cause xcode to highlight the test calling code as in error (rather than somewhere in this library). If someone supplies the parameters then xcode will highlight the error as coming from a different place in the code base.